Tests/TSCBasicTests/LazyCacheTests.swift

```diff
@@ -13,20 +13,19 @@ import XCTest
 import TSCBasic
 
 class LazyCacheTests: XCTestCase {
-    func testBasics() {
-        class Foo {
-            var numCalls = 0
-            
-            var bar: Int { return barCache.getValue(self) }
-            var barCache = LazyCache<Foo, Int>(someExpensiveMethod)
-            func someExpensiveMethod() -> Int {
-                numCalls += 1
-                return 42
-            }
-                
+    private class Foo {
+        var numCalls = 0
+
+        var bar: Int { return barCache.getValue(self) }
+        var barCache = LazyCache<Foo, Int>(someExpensiveMethod)
+        func someExpensiveMethod() -> Int {
+            numCalls += 1
```

Comment:
is this thread safe?

Reply:
Yeah I think so. The `someExpensiveMethod()` is protected by a lock. Maybe `LazyCache` should be named something like `ThreadSafeLazyCache`?